File: test/FillStyle/test08.c
#include <stdlib.h>
#include <stdio.h>
#include <errno.h>
#include <libming.h>

int main()
{
	SWFMovie m = newSWFMovieWithVersion(8);

	SWFGradient gradient = newSWFGradient();
	SWFFillStyle fill = newSWFGradientFillStyle(gradient,
		SWFFILL_LINEAR_GRADIENT);
	SWFShape shape = newSWFShape();

	SWFGradient_addEntry(gradient, 0, 255, 0, 0, 255);
	SWFGradient_addEntry(gradient, 0.25, 128, 32, 32, 255);
	SWFGradient_addEntry(gradient, 0.8, 64, 64, 64, 255);
	SWFShape_setRightFillStyle(shape, fill);
	
	SWFShape_setLine(shape, 1, 0,0,0,255);
	SWFShape_drawLine(shape, 100, 0);
	SWFShape_drawLine(shape, 0, 100);
	SWFShape_drawLine(shape, -100, 0);
	SWFShape_drawLine(shape, 0, -100);

	SWFMovie_add(m, (SWFBlock)shape);
	SWFMovie_save(m, "test08.swf");
	return 0;

}

File: test/FillStyle/test08.php
#!/usr/bin/php
<?php
$m = new SWFMovie(8);

$shape = new SWFShape();
$gradient = new SWFGradient();
$gradient->addEntry(0, 255, 0, 0, 255);
$gradient->addEntry(0.25,0x80,0x20,0x20,0xff);
$gradient->addEntry(0.8,0x40,0x40,0x40,0xff);
$fill = $shape->addGradientFill($gradient,SWFFILL_LINEAR_GRADIENT);
#$fill->moveTo(-163, -163); # this is added by swftoscript, but is wrong
$shape->setLine(1,0,0,0,255);
$shape->setRightFill($fill);
$shape->drawLine(100, 0);
$shape->drawLine(0, 100);
$shape->drawLine(-100, 0);
$shape->drawLine(0, -100);

$m->add($shape);

$m->save("test08.swf");
?>
